packages/@ionic/cli/src/lib/project/angular/index.ts

```diff
@@ -79,6 +91,20 @@ export class AngularProject extends Project {
     return false;
   }
 
+  async getWorkspaceDefaultProject(): Promise<string> {
+    try {
+      const angularJsonPath = resolve(this.directory, 'angular.json');
+      const angularWorkspace: WorkspaceSchema = await readJson(angularJsonPath);
+      if (angularWorkspace && angularWorkspace.defaultProject && angularWorkspace.defaultProject !== '') {
+        return Promise.resolve(angularWorkspace.defaultProject);
+      }
+    } catch (e) {
+      // ignore
+    }
+
+    return Promise.resolve('app');
```

Comment:
Note this MR still keeps the hardcoded default fallback of `app`, in the angular world, this should have actually been the first defined project, but this would break build scripts for people that rely on this fallback. Maybe this can be marked as a deprecated "feature" and a future major release could use the proper fallback.